esiclient/v1/node_network.py

```diff
@@ -74,31 +74,34 @@ def take_action(self, parsed_args):
                                   if node.uuid == port.node_uuid), None).name
 
             neutron_port_id = port.internal_info.get('tenant_vif_port_id')
+            network = None
 
             if neutron_port_id:
                 neutron_port = next((np for np in neutron_ports
                                      if np.id == neutron_port_id), None)
 
                 network_id = neutron_port.network_id
```

Comment:
Ah, the reason goes back to the original reason for this PR; there are cases where the neutron_port_id is specified, but the neutron_port doesn't exist. In such cases, the neutron_port is set to None, and we get this exact error. So now you just have to add a check here to only continue if neutron_port is not None.

Reply:
Ah, okay!